spec/src/main/asciidoc/overview.asciidoc

```diff
@@ -19,29 +19,33 @@
+When using a reactive model, execution is cut into small units of work that are chained together to assemble a reactive pipeline. The context under which each unit of work executes is often unpredictable and depends on the particular reactive engine used. Some units might run with the context of a thread that awaits completion, or the context of a previous unit that completed and triggered the dependent unit, or with no/undefined context at all. Existing solutions for transferring thread context, such as the EE Concurrency Utilities `ContextService`, are tied to a specific asynchrony model, promotes usage of thread pools, is difficult to use and require a lot of boilerplate code. This specification makes it possible for thread context propagation to easily be done in a type-safe way, keeping boilerplate code to a minimum, as well as allowing for thread context propagation to be done automatically for many types of reactive models.
 
-It is also important that a `CompletableFuture` and its dependent stages, and dependent stages created by those stages, and so on, all run on threads that are known by the container and properly managed by it. This specification makes it possible to do so by providing an API by which the user obtains managed `CompletableFuture` instances from a managed executor.
+We distinguish two main use-cases for propagating contexts to reactive pipelines:
+
+    - Splitting units of work into a sequential pipeline where each unit will be executed after the other. Turning an existing blocking request into an async request would produce such pipelines.
+
+    - Fanning out units of work to be executed in parallel on a managed thread pool. Launching an asynchronous job from a request without waiting for its termination would produce such pipelines.
 
 Goals
 
-    - Proper management of `CompletableFuture` threads by the container.
+    - Pluggable context propagation to the most common unit of work types.
 
-    - Mechanism for thread context propagation to `CompletableFuture` actions that reduces the need for boilerplate code.
+    - Mechanism for thread context propagation to `CompletableFuture` and `CompletionStage` units of work that reduces the need for boilerplate code.
 
-    - Full compatibility with EE Concurrency spec, such that proposed interfaces can eventually be seamlessly merged into EE Concurrency.
+    - Full compatibility with EE Concurrency spec, such that proposed interfaces can seamlessly work alongside EE Concurrency, without depending on it.
 
 === Solution
 
-This specification introduces two interfaces that contain methods that we hope will eventually be added to Jakarta EE Concurrency.
+This specification introduces two interfaces that contain methods that can work alongside EE Concurrency, if available.
```

Comment:
This pretty much repeats what's in the previous sentence, I'd just drop it and say there are two main entry points (interfaces) this spec offers.

Reply:
It's repetitive indeed. But I don't think it harms either. WDYT @njr-11 ?

Reply:
I'd say we do need something here to introduce the two main interface classes that are described in the subsequent paragraphs.  The current sentence seems fine for that and the duplication doesn't bother me, but if Matej has an idea for a more concise way of stating it, I would alternately be fine with that.